migen/build/platforms/sinara/humpback.py

```diff
@@ -0,0 +1,114 @@
+from migen.build.generic_platform import *
+from migen.build.lattice import LatticePlatform
+
+
+_io = [
+       ("clk25", 0, Pins("K9"), IOStandard("LVCMOS33")),
+
+       ("user_led", 0, Pins("H3"), IOStandard("LVCMOS33")),
+
+       ("serial0", 0,
+        Subsignal("rx", Pins("M13")),
+        Subsignal("tx", Pins("T11"), Misc("PULLUP")),
+        Subsignal("cts", Pins("T10"), Misc("PULLUP")),
+        Subsignal("rts", Pins("M15"), Misc("PULLUP")),
+        IOStandard("LVCMOS33"),
+        ),
+
+       ("serial1", 0,
+        Subsignal("rx", Pins("M11")),
+        Subsignal("tx", Pins("T13"), Misc("PULLUP")),
+        Subsignal("cts", Pins("B16"), Misc("PULLUP")),
+        Subsignal("rts", Pins("A6"), Misc("PULLUP")),
+        IOStandard("LVCMOS33"),
+        ),
+
+       ("spi", 0,
+        Subsignal("cs_n", Pins("R2")),
+        Subsignal("miso", Pins("T2")),
+        Subsignal("mosi", Pins("N5")),
+        Subsignal("clk", Pins("C8")),
+        IOStandard("LVCMOS33"),
+        ),
+
+       ("i2c", 0,
+        Subsignal("sda", Pins("T16"), Misc("PULLUP")),
+        Subsignal("scl", Pins("M12"), Misc("PULLUP")),
+        IOStandard("LVCMOS33"),
+        ),
+]
+
+
+_io += [("sw", i, Pins(p), IOStandard("LVCMOS33"))
+        for i, p in enumerate(
+            "L16 K16 L11 T14 P7 N7 T8 P6 N6 T6 R6 P5".split())]
+
+
+_connectors = [
+    ("eem0", {
```

Comment:
Maybe name them differently depending if it's an upstream or downstream connector?

Reply:
I'd leave it like that. The upstream/downstream difference merely expresses itself in non-fpga related signals (power and i2c).